### DISCUSSION TITLE
Kafka listener is not supporting json dump from python27

### DISCUSSION THREAD

**Discussion by arun6894** [miguelgrinberg/python-socketio#1132] [Q&A]
@miguelgrinberg , I am producing the same message structure which sends from server to client via message queue from the python27 file so that the client can emit it in the same way.

If I produce the structure for eg., `{'method': 'emit', 'event': 'message', 'data': 'hello- message from socket server', 'namespace': '/check', 'room': None, 'skip_sid': None, 'callback': None}`  in Kafka python3. I can see the event is triggering the client.

I tried the same with Python27 as per my requirement. I see it is not working. From python27 I tried to produce the value with json.dumps(), pickle.dumps(). both are not working.

As per the library in python3 , I see it is using pickle.dumps(). I could see this worked for me as per the conversion from python27.

```
def _listen(self):
        for message in self._kafka_listen():
            if message.topic == self.channel:
                if isinstance(message.value,bytes):
                    yield json.loads(json.loads(message.value.decode()))
                else:
                    yield pickle.loads(message.value)
```

I don't want to change the inbuilt library method structure. Is there any argument to make this possible without changing the **kafka_manager.py** file?

**Comment by miguelgrinberg** [comment]
I'm not really sure what the problem can be, but based on your description my impression is that you need to debug your python3 and python2 that write to the queue to figure out what is their difference. The queue itself shouldn't care that you are using Python 2.7 as long as you write a correct payload.

**Reply by arun6894** [reply]
I understand that there is different set of data format which is not supported with python27. The same works with python3 . I found a solution with the above snippet. Is it a good idea to make the change in inbuilt to support python27 pickle.dumps() which comes in bytes ?

**Reply by miguelgrinberg** [reply]
I don't offer support for Python 2.7 in this package, so I will not be adding any features for that version of Python.

**Reply by arun6894** [reply]
I have resolved it from the python2 code. Thanks
